File: src/zoo/Animals.java
package zoo;
public class Animals {
	public String name;
	
	public Animals(String name) {
		this.name = name;
	}
	
	public void makeNoise() {
		System.out.println("My type is " + this.getClass() + " " + this.name + " Making Noise");
	}
	
	public void sleep() {
		System.out.println("My type is " + this.getClass() + " " +  this.name + " Going to sleep");
	}
	
	public void eat() {
		System.out.println("My type is " + this.getClass() + " " + this.name + " Eating my meal");
	}
	
	public void roam() {
		System.out.println("My type is " + this.getClass() + " " + this.name + " Moving around");
	}
	
	public void wakingUp() {
		System.out.println("My type is " + this.getClass() + " " + this.name + "," + "good morning");
	}
}

File: project 2/python-code/Animals.py
#!/usr/bin/env python
# coding: utf-8

# In[5]:

#  Strategy pattern is applied

import abc
class makeNoise():
    def Noise(self): pass

class hissNoise(makeNoise):
    def Noise(self):
        return "hiss"

class growlNoise(makeNoise):
    def Noise(self):
        return "growl"

# sound variable is used to get the noise sound of different animals
class Animals:

    sound = None
    
    def __init__(self, name):
        self.name = name


    def makeNoise(self):
        print("My type is ", self.__class__.__name__, self.name, self.sound.Noise())

    def sleep(self):
        print("My type is ", self.__class__.__name__, self.name, " Going to sleep")
        
    def eat(self):
        print("My type is ", self.__class__.__name__, self.name, " Eating my meal")

    def roam(self):
        print("My type is ", self.__class__.__name__, self.name, " Moving around")

    def wakingUp(self):
        print("My type is ",  self.__class__.__name__, self.name, "good morning")

